plugins/auth-backend-module-vmware-cloud-provider/src/authenticator.ts

```diff
@@ -209,23 +222,27 @@ export const vmwareCloudAuthenticator = createOAuthAuthenticator<
   },
 
   async authenticate(input, ctx) {
-    return ctx.helper.authenticate(input).then(result => ({
+    const result = await ctx.helper.authenticate(input);
+    await jwtVerify(result.session.accessToken, ctx.jwks);
```

Comment:
I see what you're saying, but I think there's some precedent in the fact that the `openid-client` strategy [does this kind of validation](https://github.com/panva/node-openid-client/blob/563d9a347a2ddd9aac48ce0d7019f4d5c247864d/lib/client.js#L1071) immediately upon receiving an ID token. It seems a little paranoid, though I guess it could defend against a possible man-in-the-middle attack? Maybe reasonable to think of it as a sanity check, like "you just gave me this token, so I'll just double check that it's actually signed by one of the keys you say to trust"?

Are you concerned about performance impact/compute resource saturation, or more about maintenance complexity?

Reply:
Performance, maintenance, and false sense of security. The verification certificates come from the same source as the tokens right? If there's a mitm attack they can replace the certs too. You'd need to get the certs from a separate source or configure them statically instead.

Reply:
yeah, you're right. I'll close this PR and open a separate one with just the scope fix.